src/bin/hexyl.rs

```diff
@@ -140,27 +190,108 @@ fn main() {
 fn parse_hex_or_int(n: &str) -> Option<u64> {
+    let n = n.trim_start_matches('+');
     if n.starts_with("0x") {
         u64::from_str_radix(n.trim_start_matches("0x"), 16).ok()
     } else {
         n.parse::<u64>().ok()
     }
 }
+
+fn parse_range(range_raw: &str) -> Result<(u64, u64)> {
+    match range_raw.split(':').collect::<Vec<&str>>()[..] {
+        [offset_raw, bytes_to_read_raw] => {
+            let offset = parse_hex_or_int(&offset_raw).unwrap_or_else(u64::min_value);
+            let bytes_to_read = match parse_hex_or_int(bytes_to_read_raw) {
+                Some(num) if bytes_to_read_raw.starts_with('+') => num,
+                Some(num) if offset <= num => num - offset,
+                Some(num) => {
+                    return Err(format!(
+                        "cannot start reading at {} and stop reading at {}",
+                        offset, num
+                    )
+                    .into())
+                }
+                None if bytes_to_read_raw != "" => return Err("unable to parse range".into()),
+                None => u64::max_value(),
+            };
+            Ok((offset, bytes_to_read))
+        }
+        [offset_raw] => {
+            let offset = parse_hex_or_int(&offset_raw).unwrap_or_else(u64::min_value);
+            let bytes_to_read = u64::max_value();
+            Ok((offset, bytes_to_read))
+        }
+        _ => Err("expected single ':' character".into()),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn parse_empty_range() {
+        let (offset, bytes_to_read) = parse_range(":").expect("Not allowed to fail test.");
+        assert_eq!(offset, u64::min_value());
+        assert_eq!(bytes_to_read, u64::max_value());
+    }
+
+    #[test]
+    fn parse_starting_offset() {
+        let (offset, bytes_to_read) = parse_range("0x200:").expect("Not allowed to fail test.");
+        assert_eq!(offset, 512);
+        assert_eq!(bytes_to_read, u64::max_value());
+
+        let (offset, bytes_to_read) = parse_range("0x200").expect("Not allowed to fail test.");
+        assert_eq!(offset, 512);
+        assert_eq!(bytes_to_read, u64::max_value());
+    }
+
+    #[test]
+    fn parse_ending_offset() {
+        let (offset, bytes_to_read) = parse_range(":0x200").expect("Not allowed to fail test.");
+        assert_eq!(offset, u64::min_value());
+        assert_eq!(bytes_to_read, 512);
+
+        let (offset, bytes_to_read) = parse_range(":+512").expect("Not allowed to fail test.");
+        assert_eq!(offset, u64::min_value());
+        assert_eq!(bytes_to_read, 512);
+
+        let (offset, bytes_to_read) = parse_range("512:512").expect("Not allowed to fail test.");
+        assert_eq!(offset, 512);
+        assert_eq!(bytes_to_read, 0);
+
+        let (offset, bytes_to_read) = parse_range("512:+512").expect("Not allowed to fail test.");
+        assert_eq!(offset, 512);
+        assert_eq!(bytes_to_read, 512);
+    }
+
+    #[test]
+    fn parse_bad_input() {
```

Comment:
What happens if the range arg is completely empty?
```
--range ""
```

Reply:
This is the same as providing an empty range:
```
--range ":"
```

I'll add it to the test :)